testing/scenariobuilder/todoerr_scenario.go

```diff
@@ -48,6 +51,21 @@ func (s *TodoErrScenario) ExpectLine(line string) *TodoErrScenario {
 	return s
 }
 
+// WithMetadata extends existing metadata with a multiple of key value pairs
+// expected within the `metadata` field of the TODOs
+func (s *TodoErrScenario) WithMetadata(metadata *map[string]string) *TodoErrScenario {
```

Comment:
No need to pass a `*map[string]string`. Maps in go always behave as if they're `*map[type]type` even if you pass it as `map[type]type`

Reply:
The name of the func should be `WithJSONMetadata` as it only impacts the scenario if it is in json format

Reply:
`WithMetadata` -> `WithJSONMetadata`
 `*map[string]string` -> `map[string]string